include/irrString.h

```diff
@@ -975,50 +965,29 @@ static size_t multibyteToWString(string<wchar_t>& destination, const char* sourc
 	}
 }
 
-//! Same as multibyteToWString, but the other way around
-static inline size_t wStringToMultibyte(string<c8>& destination, const core::string<wchar_t>& source)
+
+inline size_t utf8ToWString(stringw &destination, const char *source)
 {
-	return wStringToMultibyte(destination, source.c_str(), (u32)source.size());
+	std::wstring_convert<std::codecvt_utf8<wchar_t>> conv;
+	destination = conv.from_bytes(source);
+	return destination.size();
 }
 
-//! Same as multibyteToWString, but the other way around
-static inline size_t wStringToMultibyte(string<c8>& destination, const wchar_t* source)
+inline size_t utf8ToWString(stringw &destination, const stringc &source)
 {
-	const u32 s = source ? (u32)wcslen(source) : 0;
-	return wStringToMultibyte(destination, source, s);
+	return utf8ToWString(destination, source.c_str());
```

Comment:
Can a `stringc` contain embedded nulls? If so, this shouldn’t delegate but use another `from_bytes`:
`conv.from_bytes(source.begin(), sorce.end())`

Reply:
This seems to suggest it can't:

https://github.com/minetest/irrlicht/blob/5b2f1927d0ccc59173e00f3276b33e08b1ca748c/include/irrString.h#L789-L798